### PR TITLE
Get from releases

### PR DESCRIPTION
**Pull request by stone-z**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -100,37 +100,37 @@ required = [
   name = "github.com/spf13/viper"
   version = "1.0.0"
 
-[[constraint]]
+[[override]]
   name = "k8s.io/api"
-  version = "kubernetes-1.16.3"
+  version = "kubernetes-1.15.11"
```
Comment (MarcelMue) [inline Gopkg.toml]: Shouldn't be necessary these are backwards compatible.

File: pkg/project/version_bundle.go [hunk 2]
```diff
@@ -8,23 +8,8 @@ func NewVersionBundle() versionbundle.Bundle {
...
-			{
-				Component:   "etcd",
-				Description: "Updated from v3.3.13 to v3.3.15.",
-				Kind:        versionbundle.KindChanged,
-			},
-			{
-				Component:   "kubernetes",
-				Description: "Updated from v1.14.6 to v1.15.5.",
+				Component:   "kvm-operator",
+				Description: "Modified to retrieve component versions from releases",
```
Comment (MarcelMue) [inline pkg/project/version_bundle.go]: This is unused - fine as a placeholder but we need a `CHANGELOG.md`

File: service/controller/cloudconfig/master_template.go [hunk 3]
```diff
@@ -7,29 +7,28 @@ import (
...
 	"github.com/giantswarm/microerror"
-	"github.com/giantswarm/randomkeys"
 
 	"github.com/giantswarm/kvm-operator/service/controller/key"
 )
 
 // NewMasterTemplate generates a new worker cloud config template and returns it
 // as a base64 encoded string.
-func (c *CloudConfig) NewMasterTemplate(customObject v1alpha1.KVMConfig, certs certs.Cluster, node v1alpha1.ClusterNode, randomKeys randomkeys.Cluster, nodeIndex int) (string, error) {
+func (c *CloudConfig) NewMasterTemplate(data IgnitionTemplateData, node v1alpha1.ClusterNode, nodeIndex int) (string, error) {
```
Comment (MarcelMue) [inline service/controller/cloudconfig/master_template.go]: This is one part of it here but we also need to alter the places where these functions are called - or is this missing in the diff somehow?

File: pkg/project/version_bundle.go (hunk 2 above)
Comment (stone-z) [reply pkg/project/version_bundle.go:12]: Added Changelog

File: Gopkg.toml (hunk 1 above)
Comment (stone-z) [reply Gopkg.toml]: Updated to newer 1.16.8 due to vulnerable dependencies in 1.16.3

File: service/controller/cloudconfig/master_template.go (hunk 3 above)
Comment (stone-z) [reply service/controller/cloudconfig/master_template.go:16]: Yep, just hadn't gotten there yet. Has been added

File: service/controller/resource/configmap/desired.go [hunk 4]
```diff
@@ -51,7 +52,14 @@ func (r *Resource) newConfigMaps(customResource v1alpha1.KVMConfig) ([]*apiv1.Co
 			return nil, microerror.Maskf(notFoundError, fmt.Sprintf("node index for master (%q) is not available", node.ID))
 		}
 
-		template, err := r.cloudConfig.NewMasterTemplate(customResource, certs, node, keys, nodeIdx)
+		ignition := cloudconfig.IgnitionTemplateData{
+			CustomObject: customResource,
+			ClusterCerts: certs,
+			ClusterKeys:  keys,
+			// Images?
```
Comment (MarcelMue) [inline service/controller/resource/configmap/desired.go:59]: Images!

You can check `aws-operator` on how to actually get the images here. We can also hang tomorrow if you want.

Comment (stone-z) [reply service/controller/resource/configmap/desired.go:59]: I checked there but I think this works differently for KVM?

Comment (MarcelMue) [reply service/controller/resource/configmap/desired.go:59]: It should be reasonably similar :)

**Comment by tfussell** [comment]
Looking good so far. Just a failing unit test now it seems.

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+*
+!kvm-operator
+!go.mod
```
Comment (MarcelMue) [inline .dockerignore:3]: Why is a dockerignore needed?

File: service/controller/resource/configmap/desired.go [hunk 6]
```diff
@@ -45,13 +48,37 @@ func (r *Resource) newConfigMaps(customResource v1alpha1.KVMConfig) ([]*apiv1.Co
 		return nil, microerror.Mask(err)
 	}
 
+	var release *releasev1alpha1.Release
+	{
+		releaseVersion := customResource.Labels[label.ReleaseVersion]
+		releaseName := fmt.Sprintf("v%s", releaseVersion)
+		release, err = r.g8sClient.ReleaseV1alpha1().Releases().Get(releaseName, apismetav1.GetOptions{})
```
Comment (MarcelMue) [inline service/controller/resource/configmap/desired.go:55]: RBAC needs to be changed to allow us getting `Releases`

File: .dockerignore (hunk 5 above)
Comment (stone-z) [reply .dockerignore]: I'm actually not convinced that it is anymore -- removing to see what happens

File: service/controller/resource/configmap/desired.go (hunk 6 above)
Comment (stone-z) [reply service/controller/resource/configmap/desired.go:55]: Added

File: service/controller/resource/configmap/desired.go (hunk 4 above)
Comment (stone-z) [reply service/controller/resource/configmap/desired.go]: It was :)

**Comment by stone-z** [comment]
I think this is good to go. I'll do flatcar in a separate PR

File: go.mod [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	github.com/giantswarm/errors v0.2.0
+	github.com/giantswarm/k8sclient v0.2.0
+	github.com/giantswarm/k8scloudconfig/v6 v6.0.1
+	github.com/giantswarm/microendpoint v0.2.0
+	github.com/giantswarm/microerror v0.2.0
+	github.com/giantswarm/microkit v0.2.0
+	github.com/giantswarm/micrologger v0.3.1
+	github.com/giantswarm/operatorkit v0.2.0
+	github.com/giantswarm/randomkeys v0.2.0
+	github.com/giantswarm/statusresource v0.0.0-20200103225110-1c017ce979b3
```
Comment (MarcelMue) [inline go.mod:17]: We can improve here later on probably this `go.mod` doesn't look super clean :D

File: service/controller/resource/configmap/desired.go [hunk 8]
```diff
@@ -45,13 +48,37 @@ func (r *Resource) newConfigMaps(customResource v1alpha1.KVMConfig) ([]*apiv1.Co
 		return nil, microerror.Mask(err)
 	}
 
+	var release *releasev1alpha1.Release
+	{
+		releaseVersion := customResource.Labels[label.ReleaseVersion]
```
Comment (MarcelMue) [inline service/controller/resource/configmap/desired.go:53]: Nit: I would like this line and the next to be in some `key.go` but can be done in a follow up :)

**Pull request merged**